### ISSUE TITLE
Encountered oom problem

### ISSUE THREAD

**Issue by yyqxnice** [PyTorchLightning/pytorch-lightning#9604]
I have encountered an oom problem. As the program runs, the video memory continues to increase. As described in #9336 , I use import pytorch lightning 1.47, but I find that 1.47 does not solve this problem. How can I solve this problem, or will your next version update this problem ？？？？

**Comment by yyqxnice** [comment]
我程序在运行的过程中测试阶段一直在增加内存，或者是有什么地方在存储tensor或许可以将它删除
During the running process of my program, the test phase has been increasing memory, or there is a place where it is stored. Maybe tensor can delete it

**Comment by yyqxnice** [comment]
During the running process of my program, the test phase has been increasing memory, or there is a place where it is stored. Maybe tensor can delete it

**Comment by justusschock** [comment]
Hi, thanks for notifying us. 

Do you have a minimal reproduction example using the boringmodel?  Otherwise it is hard for us to pinpoint the exact issue.

Best,
Justus

**Comment by yyqxnice** [comment]
> Hi, thanks for notifying us.
> 
> Do you have a minimal reproduction example using the boringmodel? Otherwise it is hard for us to pinpoint the exact issue.
> 
> Best,
> Justus

Sorry, I may not understand what you mean, because I just started to contact this aspect, but I doubt whether such a burst memory still exists because the calculation diagram is saved?

**Comment by justusschock** [comment]
We need some code that triggers the exact issue. So that we can investigate what causes this.

**Comment by yyqxnice** [comment]
> We need some code that triggers the exact issue. So that we can investigate what causes this.



Hello, I found the reason because the gradient was back propagated and calculated during the retest, plus a line with t.no_ Grad (): you can solve the oom problem.This may help you,Thank you!

**Comment by justusschock** [comment]
What code are you running? From your description it is not exactly clear to me, where the issue is.

**Comment by yyqxnice** [comment]
> What code are you running? From your description it is not exactly clear to me, where the issue is.
> 
> Can you explain a bit further how this is related to `no_grad` in your opinion?

Hello, I released some of my code below. I don't know if it will help you. If you need to check the complete code, I can give you a link about the complete [code.](https://github.com/iffiX/PGPortfolio-pytorch)
`  def training_step(self, batch, _batch_idx):
        batch = batch[0]
        new_w = self._net(batch["X"], batch["last_w"])
        del batch["X"]
        batch["setw"](new_w[:, 1:])
        return self._init_metrics() \
            .eval(batch["y"], batch["last_w"], new_w).loss


    def training_step_end(self, training_output):
            fast_train = self._train_config["fast_train"]
            self._net.eval()
            with t.no_grad():
                batch_size = self._test_set_X.shape[0]
                new_w_list = []
                for start in range(0, batch_size, 16):
                    end = min(start + 16, batch_size)
                    new_w = self._net(self._test_set_X[start:end], self._test_set_last_w[start:end])
                    new_w_list.append(new_w)
                new_w = t.cat(new_w_list, dim=0)
                self._test_set_setw(new_w[:, 1:])
                m = self._init_metrics().eval(self._test_set_y,
                                              self._test_set_last_w,
                                              new_w)
                if m.portfolio_value == 1.0:
                    logging.error("Portfolio value is the same as start, "
                                  "check input data, "
                                  "average portfolio weights {}"
                                  .format(m.pv_vector))

                self.log_dict({
                    "test_portfolio_value": m.portfolio_value,
                    "test_log_mean": m.pv_log_mean,
                    "test_loss": m.loss,
                    "test_log_mean_free": m.pv_log_mean_no_commission,
                }, prog_bar=True)

                self._net.train()
                return training_output
                
               
`

**Comment by yyqxnice** [comment]
> What code are you running? From your description it is not exactly clear to me, where the issue is.
> 
> Can you explain a bit further how this is related to `no_grad` in your opinion?

Hello, I released some of my code below. I don't know if it will help you. If you need to check the complete code, I can give you a link about the complete [code.](https://github.com/iffiX/PGPortfolio-pytorch)
`    def training_step(self, batch, _batch_idx):
        batch = batch[0]
        new_w = self._net(batch["X"], batch["last_w"])
        del batch["X"]
        batch["setw"](new_w[:, 1:])
        return self._init_metrics() \
            .eval(batch["y"], batch["last_w"], new_w).loss


    def training_step_end(self, training_output):
            fast_train = self._train_config["fast_train"]

            self._net.eval()
            with t.no_grad():
                batch_size = self._test_set_X.shape[0]
                new_w_list = []
                for start in range(0, batch_size, 16):
                    end = min(start + 16, batch_size)
                    new_w = self._net(self._test_set_X[start:end], self._test_set_last_w[start:end])
                    new_w_list.append(new_w)
                new_w = t.cat(new_w_list, dim=0)
                self._test_set_setw(new_w[:, 1:])
                m = self._init_metrics().eval(self._test_set_y,
                                              self._test_set_last_w,
                                              new_w)
                if m.portfolio_value == 1.0:
                    logging.error("Portfolio value is the same as start, "
                                  "check input data, "
                                  "average portfolio weights {}"
                                  .format(m.pv_vector))

                self.log_dict({
                    "test_portfolio_value": m.portfolio_value,
                    "test_log_mean": m.pv_log_mean,
                    "test_loss": m.loss,
                    "test_log_mean_free": m.pv_log_mean_no_commission,
                }, prog_bar=True)

                if not fast_train:
                    new_w = self._net(self._train_set_X,
                                      self._train_set_last_w)
                    self._train_set_setw(new_w[:, 1:])
                    m = m.evaluate(self._train_set_y, self._train_set_last_w, new_w)
                    self.log("train_loss", m.loss, prog_bar=True)

                self._net.train()
                return training_output`
